crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/next_target.rs

```diff
@@ -125,13 +121,7 @@ fn next_remote_node(
             // Put b behind a that hasn't been gossiped with.
             (None, Some(_)) => Ordering::Less,
             // Randomly break ties.
-            (None, None) => {
-                if rng.gen() {
-                    Ordering::Less
-                } else {
-                    Ordering::Greater
-                }
-            }
+            (None, None) => Ordering::Equal,
```

Comment:
Invalid sorting strategy, this doesn't work when the internal algorithm might compare elements more than once and needs to get the same answer. This was consistently producing a panic.

Looks like the internal algorithm is currently quicksort, don't know what it was and it doesn't really matter because I can't really see why this is important in the first place. If we want to pick a new peer randomly then shuffle the new peer list, don't do this!